lib/utils/normalizeOptions.js

```diff
@@ -49,6 +55,22 @@ function normalizeOptions(compiler, options) {
   options.client.path = `/${
     options.client.path ? options.client.path.replace(/^\/|\/$/g, '') : 'ws'
   }`;
+
+  options.devMiddleware = options.devMiddleware || {};
+
+  // this was previously done in createConfig, but it should be done
+  // here for CLI/API uniformity
+  if (!options.devMiddleware.publicPath) {
```

Comment:
I thought this was a good time to move `publicPath` handling from `createConfig` into `normalizeOptions` so it is done both on the CLI and API. Let me know if you disagree, as it is a minor breaking change for API.

Reply:
Yes, let's think about it in other PRs